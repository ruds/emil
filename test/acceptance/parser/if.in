if foo then bar else baz;
if if foo then bar else baz then quux else waldo;
if foo then if bar then baz else quux else waldo;
if foo then bar else if baz then quux else waldo;
if then bar else baz; (* error: missing condition *)
if foo then else baz; (* error: missing true branch *)
if foo then bar else; (* error: missing false branch *)
if foo bar else baz; (* error: missing then *)
if foo then bar; (* error: missing else *)
