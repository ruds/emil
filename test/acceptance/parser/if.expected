0001 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (IfExpr
                     :cond
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "666F6F"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                     :true_branch
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "626172"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                     :false_branch
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "62617A"
                         :is_op
                         false
                         :is_prefix_op
                         false))
                 :rec
                 false))
             :explicit_type_vars
             ())))
0002 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (IfExpr
                     :cond
                     (IfExpr
                         :cond
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "666F6F"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                         :true_branch
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "626172"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                         :false_branch
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "62617A"
                             :is_op
                             false
                             :is_prefix_op
                             false))
                     :true_branch
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "71757578"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                     :false_branch
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "77616C646F"
                         :is_op
                         false
                         :is_prefix_op
                         false))
                 :rec
                 false))
             :explicit_type_vars
             ())))
0003 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (IfExpr
                     :cond
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "666F6F"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                     :true_branch
                     (IfExpr
                         :cond
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "626172"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                         :true_branch
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "62617A"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                         :false_branch
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "71757578"
                             :is_op
                             false
                             :is_prefix_op
                             false))
                     :false_branch
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "77616C646F"
                         :is_op
                         false
                         :is_prefix_op
                         false))
                 :rec
                 false))
             :explicit_type_vars
             ())))
0004 (DeclTopDecl
         :decls
         ((ValDecl
             :bindings
             ((ValBind
                 :pat
                 (IdentifierPattern
                     :qualifiers
                     ()
                     :identifier
                     "6974"
                     :is_op
                     false
                     :is_prefix_op
                     false)
                 :expr
                 (IfExpr
                     :cond
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "666F6F"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                     :true_branch
                     (IdentifierExpr
                         :qualifiers
                         ()
                         :identifier
                         "626172"
                         :is_op
                         false
                         :is_prefix_op
                         false)
                     :false_branch
                     (IfExpr
                         :cond
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "62617A"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                         :true_branch
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "71757578"
                             :is_op
                             false
                             :is_prefix_op
                             false)
                         :false_branch
                         (IdentifierExpr
                             :qualifiers
                             ()
                             :identifier
                             "77616C646F"
                             :is_op
                             false
                             :is_prefix_op
                             false)))
                 :rec
                 false))
             :explicit_type_vars
             ())))
0005 ERROR
0006 ERROR
0007 ERROR
0008 ERROR
0009 ERROR
0010 (EndOfFileTopDecl)
